processor/tailsamplingprocessor/and_helper.go

```diff
@@ -0,0 +1,54 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tailsamplingprocessor // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/tailsamplingprocessor"
+
+import (
+	"fmt"
+
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/tailsamplingprocessor/internal/sampling"
+)
+
+func getNewAndPolicy(logger *zap.Logger, config AndCfg) (sampling.PolicyEvaluator, error) {
+	var subPolicyEvaluators []sampling.PolicyEvaluator
+	for i := range config.SubPolicyCfg {
+		policyCfg := config.SubPolicyCfg[i]
+		policy, _ := getAndSubPolicyEvaluator(logger, &policyCfg)
+		subPolicyEvaluators = append(subPolicyEvaluators, policy)
+	}
+	return sampling.NewAnd(logger, subPolicyEvaluators), nil
+}
+
+// Return instance of and sub-policy
+func getAndSubPolicyEvaluator(logger *zap.Logger, cfg *AndSubPolicyCfg) (sampling.PolicyEvaluator, error) {
+	switch cfg.Type {
```

Comment:
This might be the second or third time we have something like this. Perhaps it's time to refactor this to start reusing?

Reply:
I tried to do some refactoring in this area, but it started to require more changes than I initially thought. If we are going to deprecate this component, maybe it's not worth the effort. What do you think?

Reply:
Not sure we are going to deprecate this that soon. If we ever do it, we'd split this part on its own processor, so, this code will remain. Perhaps it could be done in a follow-up PR?

Reply:
sound good